backend/s3/s3.go

```diff
@@ -2992,19 +2993,13 @@ func (f *Fs) NewObject(ctx context.Context, remote string) (fs.Object, error) {
 
 // Gets the bucket location
 func (f *Fs) getBucketLocation(ctx context.Context, bucket string) (string, error) {
-	req := s3.GetBucketLocationInput{
-		Bucket: &bucket,
-	}
-	var resp *s3.GetBucketLocationOutput
-	var err error
-	err = f.pacer.Call(func() (bool, error) {
-		resp, err = f.c.GetBucketLocation(&req)
-		return f.shouldRetry(ctx, err)
+	region, err := s3manager.GetBucketRegion(ctx, f.ses, bucket, "", func(r *request.Request) {
+		r.Config.S3ForcePathStyle = aws.Bool(f.opt.ForcePathStyle)
```

Comment:
Is this necessary if we use f.ses which will already be configured with path style?

Reply:
It is necessary because `GetBucketRegion` forces it to true:

https://github.com/aws/aws-sdk-go/blob/main/service/s3/s3manager/bucket_region.go#L108

Just to be safe I added a log in the callback:

```go
	region, err := s3manager.GetBucketRegion(ctx, f.ses, bucket, "", func(r *request.Request) {
		log.Printf("Live S3ForcePathStyle=%v", *r.Config.S3ForcePathStyle)
		log.Printf("Configured S3ForcePathStyle=%v", f.opt.ForcePathStyle)
	})
```

Got:
```
2022/10/31 07:24:03 Live S3ForcePathStyle=true
2022/10/31 07:24:03 Configured S3ForcePathStyle=false
```

Reply:
Nice - thanks for checking